snuba/settings.py

```diff
@@ -24,6 +24,11 @@
 CLICKHOUSE_HTTP_PORT = int(os.environ.get("CLICKHOUSE_HTTP_PORT", 8123))
 CLICKHOUSE_MAX_POOL_SIZE = 25
 
+# Overrides the default values for the specified dataset
+CLICKHOUSE_HOST_BY_DATASET: Mapping[str, str] = {}
+CLICKHOUSE_PORT_BY_DATASET: Mapping[str, int] = {}
+CLICKHOUSE_HTTP_PORT_BY_DATASET: Mapping[str, int] = {}
+
```

Comment:
could we please have a Mapping[str, NamedTuple] like:
```
CLICKHOUSE_DATASET_CONNECTIONS: {
   "events": ClickhouseConnectionConfig(host="asdasd", port=1234),
   "events": ClickhouseConnectionConfig(host="asdasd", port=1234),
}
CLICKHOUSE_DEFAULT_CONNECTION = ClickhosueConnectionConfig(....)
```

This will make it structurally impossible to have an invalid configuration where the dataset is present and the port is missing. At the same time it will make it easier to add new parameters if needed.